### DISCUSSION TITLE
2 models need to relate to each other via different fields

### DISCUSSION THREAD

**Discussion by DiceNameIsMy** [collerek/ormar#960] [Q&A] [answered]
Here you can see that the Basket can have multiple Items related to it, and it also has an ID of the first item that has been put there. 

I couldn't find how to relate them to each other in documentation, Q&A section, and via google search. I hope someone can help me out.

```
class Basket(ormar.Model):
    id: int = ormar.Integer(primary_key=True)
    initial_item: Item = ormar.ForeignKey(Item, nullable=True)


class Item(ormar.Model):
    id: int = ormar.Integer(primary_key=True)
    name: str = ormar.String(max_length=255)
    basket: Basket = ormar.ForeignKey(Basket)
```

**Comment by alexol91** [comment, marked as answer]
In my opinion, this problem have two diferents ways to by resolved.

1. Define the relationship names, so that Ormar understands that you are referring to two different relationships.  That is, by saying that an item belongs to a basket, you create a relationship, but it must not conflict with the reverse of initial_item, which has a reverse link to the basket for which it is first, so the item variable (for all items) is called basket, but the related_name of initial_item is called “for_basket”

`class Basket(ormar.Model):
    id: int = ormar.Integer(primary_key=True)
    initial_item: Item = ormar.ForeignKey(Item, related_name="for_basket", nullable=True)


class Item(ormar.Model):
    id: int = ormar.Integer(primary_key=True)
    name: str = ormar.String(max_length=255)
    basket: Basket = ormar.ForeignKey(Basket, related_name="items") `

2. If this doesn't work for you, or you want to have a simpler life... You can create a field is_initial, Boolean that is True only for the "initial_item", and filter with a class method, or in the where of the query
`class Basket(ormar.Model):
    id: int = ormar.Integer(primary_key=True)

class Item(ormar.Model):
    id: int = ormar.Integer(primary_key=True)
    name: str = ormar.String(max_length=255)
    initial_item: bool = ormar.Boolean(default=False)
    basket: Basket = ormar.ForeignKey(Basket, related_name="items") `

I hope it helps you!

**Reply by DiceNameIsMy** [reply]
There is a problem with the first solution: In the Basket class, Item is not yet defined so it's not possible to make a relation.

I have solved this issue already with a different data structure, but thanks for the reply anyways.
